In [5]:
import pandas as pd
import numpy as np

In [6]:
df = pd.read_csv(f'../processed_data/Senate25-Tasmania-polling-place-votes.csv')

In [7]:
df

,PollingPlaceNm,GreensVotes,TotalVotes,GreensPercentage,DivisionNm,Redistribution,LegCo,LocalCouncil,PremisesAddress,PremisesNm,Latitude,Longitude
0,Beaconsfield,95,923,10.3,Bass,Bass,Rosevears,West Tamar,"92 Weld St, BEACONSFIELD 7270",Beaconsfield Community Centre,-41.202993,146.818608
1,Beauty Point,49,551,8.9,Bass,Bass,Rosevears,West Tamar,"Beatty St, BEAUTY POINT 7270",Beauty Point Uniting Church Hall,-41.148983,146.817506
2,Branxholm,19,198,9.6,Bass,Bass,McIntyre,Dorset,"Stoke St, BRANXHOLM 7261",Branxholm Community Hall,-41.165606,147.736480
3,Bridport,67,970,6.9,Bass,Bass,McIntyre,Dorset,"Main St, BRIDPORT 7262",Bridport Hall,-41.002600,147.395000
4,Derby,30,105,28.6,Bass,Bass,McIntyre,Dorset,"1 Krushka St, DERBY 7264",Derby Town Hall,-41.149330,147.802360
...,...,...,...,...,...,...,...,...,...,...,...,...
276,Triabunna,36,547,6.6,Lyons,Franklin,Prosser,Glamorgan-Spring Bay,"3 Vicary St, TRIABUNNA 7190",Triabunna Hall,-42.508220,147.910384
277,Tunnack,7,208,3.4,Lyons,Lyons,Prosser,Southern Midlands,"2147 Tunnack Rd, TUNNACK 7120",Tunnack Public Hall,-42.456521,147.461779
278,Westbury,128,1079,11.9,Lyons,Lyons,McIntyre,Meander Valley,"26 Lyall St, WESTBURY 7303",Westbury Town Hall,-41.528106,146.831020
279,Westerway,46,346,13.3,Lyons,Lyons,Derwent,Derwent Valley,"1619 Gordon River Rd, WESTERWAY 7140",Westerway Primary School,-42.677500,146.787000


In [10]:
def breakdown(year, category):
    """Breakdown the senate vote"""

    assert category in ['DivisionNm', 'Redistribution', 'LegCo', 'LocalCouncil']
    
    df = pd.read_csv(f'../processed_data/Senate{year}-Tasmania-polling-place-votes.csv')
    
    divisions = df[category].dropna().unique()
    divisions = np.sort(divisions)

    for div in divisions:
        greens_votes = df[df[category] == div]['GreensVotes'].sum()
        total_votes = df[df[category] == div]['TotalVotes'].sum()
        greens_pct = (greens_votes / total_votes) * 100
        print(div, f'{greens_pct:.2f}%')

In [11]:
breakdown(25, 'LegCo')

Derwent 10.33%
Elwick 24.87%
Hobart 42.34%
Huon 25.23%
Launceston 16.65%
McIntyre 10.72%
Mersey 8.64%
Montgomery 9.51%
Murchison 9.89%
Nelson 29.03%
Pembroke 21.84%
Prosser 14.52%
Rosevears 15.61%
Rumney 17.73%
Windermere 11.68%


In [12]:
breakdown(25, 'LocalCouncil')

Break O'Day 16.79%
Brighton 8.82%
Burnie City 9.93%
Central Coast 9.83%
Central Highlands 6.41%
Circular Head 7.96%
Clarence City 20.56%
Derwent Valley 10.53%
Devonport City 8.96%
Dorset 7.67%
Flinders 18.08%
George Town 8.77%
Glamorgan-Spring Bay 13.38%
Glenorchy City 19.22%
Hobart City 38.21%
Huon Valley 23.05%
Kentish 9.74%
King Island 6.56%
Kingborough 26.58%
Latrobe 8.06%
Launceston City 15.39%
Meander Valley 11.94%
Northern Midlands 8.29%
Sorell 20.02%
Southern Midlands 7.14%
Tasman 22.28%
Waratah-Wynyard 10.64%
West Coast 6.38%
West Tamar 15.30%


In [13]:
breakdown(25, 'DivisionNm')

Bass 14.55%
Braddon 9.28%
Clark 30.40%
Franklin 22.69%
Lyons 11.67%


In [14]:
breakdown(25, 'Redistribution')

Bass 14.62%
Braddon 9.28%
Clark 31.92%
Franklin 17.93%
Lyons 13.27%


Senate25 - New electorates (swing from old boundaries):
- Clark 31.92% (+1.52%)
- Franklin 17.93% (-4.76%)
- Bass 14.62% (+0.07%)
- Lyons 13.27% (+1.6%)
- Braddon 9.28% (-)

In [18]:
breakdown(22, 'DivisionNm')

Bass 13.26%
Braddon 7.40%
Clark 28.78%
Franklin 20.66%
Lyons 10.30%


In [19]:
breakdown(22, 'Redistribution')

Bass 13.38%
Braddon 7.40%
Clark 30.12%
Franklin 16.28%
Lyons 11.60%


Senate22 - New electorates (swing from old boundaries):
- Clark 30.12% (+1.34%)
- Franklin 16.28% (-4.38%)
- Bass 13.38% (+0.12%)
- Lyons 11.60% (+1.30%)
- Braddon 7.40% (-)